tools/pythonpkg/src/python_udf.cpp

```diff
@@ -338,18 +334,17 @@ static scalar_function_t CreateNativeFunction(PyObject *function, PythonExceptio
 					                            exception.what());
 				} else if (exception_handling == PythonExceptionHandling::RETURN_NULL) {
 					PyErr_Clear();
-					ret = Py_None;
+					FlatVector::SetNull(result, row, true);
+					continue;
 				} else {
 					throw NotImplementedException("Exception handling type not implemented");
 				}
 			} else if ((!ret || ret == Py_None) && default_null_handling) {
 				throw InvalidInputException(NullHandlingError());
 			}
-			python_objects.push_back(py::reinterpret_steal<py::object>(ret));
-			python_results[row] = ret;
+			TransformPythonObject(ret, result, row);
 		}
 
-		NumpyScan::ScanObjectColumn(python_results.data(), sizeof(PyObject *), input.size(), 0, result);
```

Comment:
Why don't we replace the call inside of `ScanObjectColumn`? I doubt we have enough tests with udfs to cover these changes

Reply:
`ScanNumpyObject` also uses the new method - I figured there was no reason to materialize the entire vector into Python objects only to then convert, given that the conversion is done one at a time anyway. This approach allows us to destroy the objects in between function calls which can reduce memory consumption in case we are returning long lists.